### PR TITLE
Fix issue 282

### PR DESCRIPTION
**Pull request by aparzi**
Hi @assuncaocharles,
this PR fix issue #282. 

The main problem was how the connection to the DB was opened. Specifically it was open within the "map" function and this on large numbers of insertion led to exhaust the resources.

### CODE REVIEW

**Comment by aparzi** [comment]
Hi @assuncaocharles,
any update for this PR?

Thanks

File: projects/playground/src/app/app.component.ts [hunk 1]
```diff
@@ -30,20 +30,20 @@ export class AppComponent {
...
-        },
-        {
-          name: `charles number ${Math.random() * 10}`,
-          email: `email number ${Math.random() * 10}`,
-        },
-      ])
-      .subscribe((result) => {
-        console.log('result add: ', result);
+    const randomData: Array<any> = [];
+    for (let i = 0; i < 200000; i++) {
```
Comment (assuncaocharles) [inline projects/playground/src/app/app.component.ts:34]: Magic number ?

File: projects/playground/src/app/app.component.ts [hunk 2]
```diff
@@ -30,20 +30,20 @@ export class AppComponent {
...
-          email: `email number ${Math.random() * 10}`,
-        },
-        {
-          name: `charles number ${Math.random() * 10}`,
-          email: `email number ${Math.random() * 10}`,
-        },
-      ])
-      .subscribe((result) => {
-        console.log('result add: ', result);
+    const randomData: Array<any> = [];
```
Comment (assuncaocharles) [inline projects/playground/src/app/app.component.ts:33]: We shouldn't have any

File: projects/playground/src/app/app.component.ts (hunk 1 above)
Comment (assuncaocharles) [reply projects/playground/src/app/app.component.ts:34]: Ha gotcha, only example

**Comment by assuncaocharles** [comment]
Hi @aparzi  

Thanks for opening it, I will test it locally today

**Comment by gbharathkumar** [comment]
Hi @assuncaocharles ,

I am trying to bulkadd 40k records and getting "data may be stale" warning and taking long time to add records. I think that's due to large number of open transactions to insert the records.

I hope the above fix for bulkadd should improve performance a lot better and hence waiting for this fix to be available. 

Can you please let us know when the fix will b merged.

**Comment by aparzi** [comment]
Hi @assuncaocharles,
were you able to test the changes?

**Comment by assuncaocharles** [comment]
Thanks @aparzi LGTM

**Comment by aparzi** [comment]
Hi @gbharathkumar,
the fix is merged in master, colud you test if bulkAdd method works.

**Comment by gbharathkumar** [comment]
Hi @aparzi ,

I did a fresh install ( yarn install) and I still don't see the latest changes reflect in the files present in my node_modules folder.

Am I missing something here. Or you need to update the minor version and publish the library to get the changes.

Can you help here.

**Comment by aparzi** [comment]
@gbharathkumar for this question it is better to ask @assuncaocharles

**Comment by gbharathkumar** [comment]
Hi @assuncaocharles,
Can you help here on why the latest changes are not getting reflected when I do a fresh yarn install.

By any chance do you need to update the version and publish it.

@aparzi,
Btw, did you able to see the latest changes when you do a yarn install

**Comment by gbharathkumar** [comment]
Hi @assuncaocharles , @aparzi 
I can see below error while publishing.

You cannot publish over the previously published versions: 9.4.0.

https://github.com/assuncaocharles/ngx-indexed-db/runs/4241527836?check_suite_focus=true

Can you please fix this

**Comment by assuncaocharles** [comment]
Hi @gbharathkumar 

The release haven't happened yet

**Comment by assuncaocharles** [comment]
@aparzi @gbharathkumar It's published now

**Comment by gbharathkumar** [comment]
Hi @assuncaocharles @aparzi
It works like a charm.
Thanks

**Pull request merged**